huaweicloud/services/acceptance/elb/data_source_huaweicloud_elb_ipgroups_test.go

```diff
@@ -0,0 +1,101 @@
+package elb
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccDatasourceIpGroups_basic(t *testing.T) {
+	rName := "data.huaweicloud_elb_ipgroups.test"
+	dc := acceptance.InitDataSourceCheck(rName)
+	name := acceptance.RandomAccResourceName()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatasourceIpGroups_basic(name),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttrSet(rName, "ipgroups.#"),
+					resource.TestCheckResourceAttrSet(rName, "ipgroups.0.name"),
+					resource.TestCheckResourceAttrSet(rName, "ipgroups.0.id"),
+					resource.TestCheckResourceAttrSet(rName, "ipgroups.0.description"),
+					resource.TestCheckResourceAttrSet(rName, "ipgroups.0.project_id"),
+					resource.TestCheckResourceAttrSet(rName, "ipgroups.0.ip_list.0.ip"),
+					resource.TestCheckResourceAttrSet(rName, "ipgroups.0.created_at"),
+					resource.TestCheckResourceAttrSet(rName, "ipgroups.0.updated_at"),
+					resource.TestCheckOutput("name_filter_is_useful", "true"),
+					resource.TestCheckOutput("ipgroup_id_filter_is_useful", "true"),
+					resource.TestCheckOutput("ip_address_filter_is_useful", "true"),
+					resource.TestCheckOutput("description_filter_is_useful", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDatasourceIpGroups_basic(name string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+data "huaweicloud_elb_ipgroups" "test" {
+  depends_on = [huaweicloud_elb_ipgroup.test]
+}
+
+data "huaweicloud_elb_ipgroups" "name_filter" {
+  depends_on = [huaweicloud_elb_ipgroup.test]
+  name       = "%[2]s"
+}
+output "name_filter_is_useful" {
+  value = length(data.huaweicloud_elb_ipgroups.name_filter.ipgroups) > 0 && alltrue(
+  [for v in data.huaweicloud_elb_ipgroups.name_filter.ipgroups[*].name :v == "%[2]s"]
+  )  
+}
+
+data "huaweicloud_elb_ipgroups" "ipgroup_id_filter" {
+  depends_on = [huaweicloud_elb_ipgroup.test]
```

Comment:
this data source has use the resource huaweicloud_elb_ipgroup.test, so it is not needed to depend it
and fix other places